gen/wrapper.jl

```diff
@@ -0,0 +1,97 @@
+# Script to parse HSL headers and generate Julia wrappers.
+using Clang
+using Clang.Generators
+using JuliaFormatter
+
+include("rewriter.jl")
+
+function wrapper(name::String, headers::Vector{String}, optimized::Bool)
+
+  @info "Wrapping $name"
+  solver = split(name, "_")[2]
+
+  cd(@__DIR__)
+  include_dir = joinpath(@__DIR__, "..", "deps", "usr", "include")
+
+  options = load_options(joinpath(@__DIR__, "hsl.toml"))
+  options["general"]["library_name"] = "lib" * name
+  options["general"]["output_file_path"] = joinpath("..", "src", "$(name).jl")
+  optimized && (options["general"]["output_ignorelist"] = ["$(solver)realtype_[sdcz]_\$",
+                                                           "$(solver)pkgtype_[sdcz]_\$",
+                                                           "$(solver)pkgtype_[il]_\$",
+                                                           "$(solver)_pkgtype_[il]_\$",
+                                                           "$(solver)cntltype_[sdcz]_\$",
+                                                           "$(solver)_control_[dczl]\$",
+                                                           "$(solver)_info_[dczl]\$",
+                                                           "$(solver)_[afs]info_d\$",
+                                                           "$(solver)_solve_control_d\$"])
+
+  args = get_default_args()
+  push!(args, "-I$include_dir")
+  
+  ctx = create_context(headers, args, options)
+  build!(ctx)
+
+  path = options["general"]["output_file_path"]
+
+  format_file(path, YASStyle())
+  rewrite!(path, name, optimized)
+  return nothing
+end
+
+function main(name::String="all"; optimized::Bool=false)
+  hsl = joinpath(@__DIR__, "..", "deps", "usr", "include")
+
+  if name == "all" || name == "hsl_ma48"
+    wrapper("hsl_ma48", ["$hsl/hsl_ma48s.h", "$hsl/hsl_ma48d.h"], optimized)
+  end
+
+  if name == "all" || name == "hsl_ma57"
+    wrapper("hsl_ma57", ["$hsl/hsl_ma57s.h", "$hsl/hsl_ma57d.h"], optimized)
+  end
+
+  if name == "all" || name == "hsl_ma77"
+    wrapper("hsl_ma77", ["$hsl/hsl_ma77s.h", "$hsl/hsl_ma77d.h"], optimized)
+  end
+
+  if name == "all" || name == "hsl_ma86"
+    wrapper("hsl_ma86", ["$hsl/hsl_ma86s.h", "$hsl/hsl_ma86d.h", "$hsl/hsl_ma86c.h", "$hsl/hsl_ma86z.h"], optimized)
```

Comment:
Isn't there a way to grab `$hsl/hsl_$(package)[sdcz].h` automatically?

Reply:
I propose a function `hsl_headers` to easily provide the variants of the headers.
I wanted to use 
```julia
[joinpath(hsl, header) for header in readdir(hsl) if endswith(header, r"hsl_ma48[sdczil].h")]
```
but if the headers are not available, we don't have an error and the interface is not generated.
I would like to avoid this case.